src/common.c

```diff
@@ -1154,6 +1154,9 @@ ssize_t ofi_bsock_flush(struct ofi_bsock *bsock)
 				   &bsock->sq.data[bsock->sq.head],
 				   avail, MSG_NOSIGNAL, bsock);
 	if (ret < 0) {
+		if (ret == -FI_EINPROGRESS_URING)
+		    return ret;
+
```

Comment:
Note: This isn't a problem with this change, but possibly another change that might be needed.

ofi_bsock_flush() is called from ep shutdown() to try forcing any buffered data out the socket.  I don't think there are any guarantees that that data will be sent, so it's mostly to provide a best effort on pushing out any last data.  We may need to modify the ep shutdown() call to use a standard socket send() instead of possibly going through uring.

Reply:
Thanks. I will address your comment in the next PR.